src/Microsoft.AspNetCore.SignalR.Client/HttpConnectionFactory.cs

```diff
@@ -27,16 +27,25 @@ public HttpConnectionFactory(IOptions<HttpConnectionOptions> options, ILoggerFac
             {
                 throw new ArgumentNullException(nameof(loggerFactory));
             }
-            
+
             _httpConnectionOptions = options.Value;
             _loggerFactory = loggerFactory;
         }
 
         public async Task<ConnectionContext> ConnectAsync(TransferFormat transferFormat, CancellationToken cancellationToken = default)
         {
             var connection = new HttpConnection(_httpConnectionOptions, _loggerFactory);
-            await connection.StartAsync(transferFormat, cancellationToken);
-            return connection;
+            try
+            {
+                await connection.StartAsync(transferFormat, cancellationToken);
+                return connection;
+            }
+            catch
+            {
+                // Make sure the connection is disposed, in case it allocated any resources before failing.
+                await connection.DisposeAsync();
```

Comment:
Should this be wrapped in its own try catch? If an error is thrown out of DisposeAsync it will hide the error that originally came from StartAsync

Reply:
This never throws 😆  (famous last words)

Reply:
A `Dispose` that throws an exception (other than an execution-engine exception like `StackOverflowException`) is an incorrectly-written `Dispose`.